src/solvers/dgsem/basis_lobatto_legendre.jl

```diff
@@ -564,35 +563,39 @@ This implements algorithm 25 "GaussLobattoNodesAndWeights" from the book
   [DOI:10.1007/978-90-481-2261-5](https://doi.org/10.1007/978-90-481-2261-5)
 """
 # From FLUXO (but really from blue book by Kopriva)
-function gauss_lobatto_nodes_weights(n_nodes::Integer)
+function gauss_lobatto_nodes_weights(RealT, n_nodes::Integer)
     # From Kopriva's book
     n_iterations = 10
-    tolerance = 1e-15
+
+    tolerance = 10 * eps(RealT)
+    if RealT == Float64
+        tolerance = 1e-15
+    end
 
     # Initialize output
-    nodes = zeros(n_nodes)
-    weights = zeros(n_nodes)
+    nodes = zeros(RealT, n_nodes)
+    weights = zeros(RealT, n_nodes)
 
     # Special case for polynomial degree zero (first order finite volume)
     if n_nodes == 1
-        nodes[1] = 0
-        weights[1] = 2
+        nodes[1] = zero(RealT)
+        weights[1] = RealT(2)
```

Comment:
```suggestion
        nodes[1] = 0
        weights[1] = 2
```
Not necessary if you assign to a vector of type `RealT`

Reply:
Same applies to other locations below, where you assign/use integers